src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/type_selection.tsx

```diff
@@ -237,6 +238,8 @@ class TypeSelection extends React.Component<TypeSelectionProps, TypeSelectionSta
     const onClick =
       'aliasUrl' in visType
```

Comment:
Personally, I find this bifurcation in logic a bit tricky to follow. What do you think of refactoring it so that `redirectToVis` contains the `'aliasUrl' in visType` condition and performs the right type of redirection? This would also deduplicate the identical `createUiStatsReporter` call, which would make it easier for someone to track it down and understand what kind of data will flow through it.

Reply:
I see what you're saying, I hadn't realized that this code was directly calling `onVisTypeSelected`.